src/Razor/src/Microsoft.CodeAnalysis.Razor.CohostingShared/Completion/CohostDocumentCompletionEndpoint.cs

```diff
@@ -153,7 +153,13 @@ public ImmutableArray<Registration> GetRegistrations(VSInternalClientCapabilitie
             // results we don't want to show. So we want to call HTML LSP only if we know we are in HTML content.
             if (documentPositionInfo.LanguageKind == RazorLanguageKind.Html)
             {
-                htmlCompletionList = await GetHtmlCompletionListAsync(request, razorDocument, razorCompletionOptions, correlationId, cancellationToken).ConfigureAwait(false);
+                var completionParams = new RazorVSInternalCompletionParams()
```

Comment:
Can/should this endpoint just use `RazorVSCompletionParams` in its definition directly? They look like they'd be serialization-compatible?

Could be a follow up though, if this needs to get into P2 under M2. We also need to make sure this doesn't do anything weird in VS Code, but that can also wait since we control insertions there more directly.

Reply:
> Can/should this endpoint just use RazorVSCompletionParams in its definition directly? They look like they'd be serialization-compatible?

I'll take a look

>  if this needs to get into P2 under M2

I'm not too concerned about getting this into P2

> We also need to make sure this doesn't do anything weird in VS Code

At some point, I'm going to have to get comfortable doing this, but for now, I think I'd still like to lean on you for that verification

Reply:
> I'll take a look

Seems to work, updated in commit 2

Reply:
> At some point, I'm going to have to get comfortable doing this, but for now, I think I'd still like to lean on you for that verification

Sure thing, but I'm sure I'll forget by the week after next when I get back to do it 😛

It's pretty easy though, just create a `.vscode` folder under your project root, and a `settings.json` file in there, and put in:
```
{
    "dotnet.server.componentPaths": {
        "razorExtension": "C:\\[Path to your Razor working folder]\artifacts\\bin\\Microsoft.VisualStudioCode.RazorExtension\\Debug\\net9.0\\"
    }
}
```